rpc.go

```diff
@@ -194,17 +195,24 @@ func (t *Topic) Publish(
 	}
 	t.lk.Lock()
 	t.ongoing[msgID] = ongoingMessage{
-		ctx:    ctx,
-		data:   data,
-		opts:   args.pubOpts,
-		respCh: respCh,
+		ctx:       ctx,
+		data:      data,
+		opts:      args.pubOpts,
+		respCh:    respCh,
+		republish: args.republish,
 	}
 	t.lk.Unlock()
 
 	if err := t.t.Publish(ctx, data, args.pubOpts...); err != nil {
 		return nil, fmt.Errorf("publishing to topic: %v", err)
 	}
 
+	if args.ignoreResponse && !args.republish {
+		t.lk.Lock()
+		delete(t.ongoing, msgID)
+		t.lk.Unlock()
+		return nil, nil
+	}
```

Comment:
Bail (don't spawn Go routine) if we don't care about the response and we're not republishing.